src/coreclr/interpreter/compiler.h

```diff
@@ -403,6 +468,35 @@ class InterpCompiler
         VarOnly = 1,
         EmbedParent = 2,
     };
+
+    enum class HelperArgType
+    {
+        GenericResolution,
+        Value
+    };
+
+    struct TokenArg
+    {
+        CORINFO_RESOLVED_TOKEN* token;
+        InterpCompiler::GenericHandleEmbedOptions options;
+    };
+
+    struct GenericHandleData
+    {
+        GenericHandleData(int genericVar, int dataItemIndex)
+            : argType(HelperArgType::GenericResolution), genericVar(genericVar), dataItemIndex(dataItemIndex) {}
+
+        GenericHandleData(int dataItemIndex)
+            : argType(HelperArgType::Value), genericVar(-1), dataItemIndex(dataItemIndex) {}
+
+        GenericHandleData() = default;
+
+        HelperArgType argType = HelperArgType::Value;
+        int genericVar = -1; // Set to a meaningful value if HelperArgType is GenericResolution
```

Comment:
Something more specific than 'a meaningful value' would be helpful here since by browsing around the vicinity I can't tell what makes the value meaningful or how I would distinguish between a valid genericVar and an invalid one. It may become obvious to me as I continue reading the PR, though!

My assumption upon first glance is it's the zero-based index of the generic parameter in the method's set of reachable generic parameters (where i use 'reachable generic parameters' to mean the set of all the enclosing type parameters + all the method parameters)

Reply:
OK, I'm at this point again and it's 'the interp var containing the generic dictionary' I think?